api/pkg/transformer/types/converter/converter.go

```diff
@@ -500,14 +501,24 @@ func ToFloat32List(val interface{}) ([]float32, error) {
 		}
 		return out, nil
 	case float32, float64:
-		val := reflect.ValueOf(v)
-		return []float32{float32(val.Float())}, nil
+		d, err := ToFloat32(val)
```

Comment:
I feel it's incorrect. NaN and nil is 2 different values. In some cases user/client library might have different handling for those.  Ideally, instead of changing how we convert NaN, we should change our cache mechanism to support NaN.

Reply:
eventually all `NaN` value will converted to `null` in our case (gota handle `NaN` as nil), that's why I modify it to `null`.  I already did 2 approaches:
* using gob instead of `json` encoding, it is surprisingly slower
* modify `NaN` (float64) value in feast row values to string before inserting to cache. It is work well but there is small latency increase.

Reply:
so the response from API is still the same using `NaN` or null

Reply:
Ok got it.